### PR TITLE
add 2 new keyboards and fixed some problem

### PR DESCRIPTION
**Pull request by duongaanh**
## Description

Added 2 new board by CherryB Studio ( CB87RGB and CB87v2)
Fixed indicator leds problem in CB1800 and CB87 boards

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cherrybstudio/cb87rgb/cb87rgb.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
+
```
Comment (drashna) [inline keyboards/cherrybstudio/cb87rgb/cb87rgb.c:60]:
```suggestion
```

File: keyboards/cherrybstudio/cb87rgb/config.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+#define RGB_DI_PIN E6
+#define DRIVER_LED_TOTAL 92
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Comment (drashna) [inline keyboards/cherrybstudio/cb87rgb/config.h:53]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
```

File: keyboards/cherrybstudio/cb87v2/config.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+#define BACKLIGHT_LEVELS 3
+#endif
+
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+#define RGB_DI_PIN E6
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/cherrybstudio/cb87v2/config.h:60]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/cherrybstudio/cb87rgb/config.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200
+#define RGB_MATRIX_LED_FLUSH_LIMIT 16
+#define RGB_MATRIX_KEYPRESSES
+
```
Comment (drashna) [inline keyboards/cherrybstudio/cb87rgb/config.h:57]: Just a heads up, rgb matrix animations need to be explicitly enabled.  They are opt in now. 

So as is, none of the animations will be enabled. 
```suggestion
// RGB Matrix Animation modes. Explicitly enabled
// For full list of effects, see:
// https://docs.qmk.fm/#/feature_rgb_matrix?id=rgb-matrix-effects
#    define ENABLE_RGB_MATRIX_ALPHAS_MODS
#    define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#    define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_BREATHING
#    define ENABLE_RGB_MATRIX_BAND_SAT
#    define ENABLE_RGB_MATRIX_BAND_VAL
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#    define ENABLE_RGB_MATRIX_CYCLE_ALL
#    define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#    define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#    define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#    define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#    define ENABLE_RGB_MATRIX_DUAL_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#    define ENABLE_RGB_MATRIX_RAINDROPS
#    define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#    define ENABLE_RGB_MATRIX_HUE_BREATHING
#    define ENABLE_RGB_MATRIX_HUE_PENDULUM
#    define ENABLE_RGB_MATRIX_HUE_WAVE
#    define ENABLE_RGB_MATRIX_PIXEL_RAIN
#    define ENABLE_RGB_MATRIX_PIXEL_FLOW
#    define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
// enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS is defined
#    define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#    define ENABLE_RGB_MATRIX_DIGITAL_RAIN
// enabled only of RGB_MATRIX_KEYPRESSES or RGB_MATRIX_KEYRELEASES is defined
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#    define ENABLE_RGB_MATRIX_SPLASH
#    define ENABLE_RGB_MATRIX_MULTISPLASH
#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cherrybstudio/cb87v2/config.h (hunk 3 above)
Comment (duongaanh) [reply keyboards/cherrybstudio/cb87v2/config.h:60]: oh this is my mistake that I forgot to listed all the needed efects !

File: keyboards/cherrybstudio/cb87rgb/cb87rgb.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    {   0,  48 }, { 13,  48 }, { 26,  48 }, { 39,  48 }, { 52,  48 }, { 65,  48 }, { 78,  48 }, { 91,  48 }, { 104,  48 }, { 117,  48 }, { 130,  48 }, { 143,  48 }, { 156,  48 }, {   169,  48 },                                 {   208,  48 },
+    {   0,  64 }, { 13,  64 }, { 26,  64 },                                        { 78,  64 },                                          { 130,  64 }, { 143,  64 }, { 156,  64 }, {   169,  64 },                 {   195,  64 }, {   208,  64 }, {   224,  64 }, {   182,  0 }
+}, {
+    // LED Index to Flag
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,       1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,    1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,       1,
+    1, 1, 1,          1,          1, 1, 1, 1, 1, 1, 1, 1
```
Comment (drashna) [inline keyboards/cherrybstudio/cb87rgb/cb87rgb.c:47]: You probably want to set the flags to properly indicate the modifier keys.

Comment (duongaanh) [reply keyboards/cherrybstudio/cb87rgb/cb87rgb.c:47]: does it nessesary because I understand those flags should work normally

Comment (drashna) [reply keyboards/cherrybstudio/cb87rgb/cb87rgb.c:47]: It is not strictly necessary, but would highly recommend doing so.

Comment (duongaanh) [reply keyboards/cherrybstudio/cb87rgb/cb87rgb.c:47]: Thank you anyway! 
But according to QMK Documentation:

> It is recommended that LEDs are set to only 1 type.

so I'm not sure should I change those bits or keep it

**Review by drashna — APPROVED** [review]

File: keyboards/cherrybstudio/cb87v2/config.h [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4342 // CB
+#define PRODUCT_ID      0x8788
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    CherryB.Studio
+#define PRODUCT         CherryB.Studio CB87v2
```
Comment (fauxpark) [inline keyboards/cherrybstudio/cb87v2/config.h:23]:
```suggestion
#define PRODUCT         CB87v2
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**